airflow/models/abstractoperator.py

```diff
@@ -494,18 +493,15 @@ def expand_mapped_task(self, run_id: str, *, session: Session) -> tuple[Sequence
                     total_length,
                 )
                 unmapped_ti.state = TaskInstanceState.SKIPPED
-                indexes_to_map = ()
             else:
-                # Otherwise convert this into the first mapped index, and create
-                # TaskInstance for other indexes.
-                unmapped_ti.map_index = 0
-                self.log.debug("Updated in place to become %s", unmapped_ti)
-                all_expanded_tis.append(unmapped_ti)
-                indexes_to_map = range(1, total_length)
-            state = unmapped_ti.state
-        elif not total_length:
+                # Delete the original task so we can create new ones
+                self.log.debug("Deleting the original task instance: %s", unmapped_ti)
+                state = unmapped_ti.state
+                session.delete(unmapped_ti)
```

Comment:
The fact that the original task instance is _converted_ instead of deleted was explicitly mentioned in the original proposal, so it would be best if we could keep the behaviour (I think there are some subtle differences but don’t recall exactly what). Would it be possible?

Reply:
If the scheduler already reached a faulty state (e.g. `[-1, 0, 1]` in your case) we can of course delete the ti to rescure it, but otherwise it’d be best to not delete things.

Reply:
in dagrun creation we explicitly check for duplicates - could do the same here.
https://github.com/apache/airflow/blob/01a819a42daa7990c30ab9776208b3dcb9f3a28b/airflow/jobs/scheduler_job.py#L1046-L1053

Reply:
Ok I'll try to make it work in a new commit.
So if I understood it correctly, what is preferred is:
- I keep the old logic that changes the `-1` map_index TI into the `0` map_index TI
- After that I detect if a `0` map_index ti already exists if so I deleted the original `-1` TI

Reply:
I update the code to have this change, please let me know if there is anything else you would like to see different